src/pushsource/_impl/backend/koji_source.py

```diff
@@ -620,34 +620,42 @@ def _get_operator_item(self, nvr, meta, archives):
         extra = meta.get("extra") or {}
         typeinfo = extra.get("typeinfo") or {}
         operator_manifests = typeinfo.get("operator-manifests") or {}
-        archive_name = operator_manifests.get("archive")
+        operator_archive_name = operator_manifests.get("archive")
         image = typeinfo.get("image") or {}
         image_operator_manifests = image.get("operator_manifests") or {}
         related_images = image_operator_manifests.get("related_images") or {}
         pullspecs = related_images.get("pullspecs") or []
         operator_related_images = [spec["new"] for spec in pullspecs if spec.get("new")]
 
-        if not archive_name:
+        if not operator_archive_name:
             # Try legacy form
-            archive_name = extra.get("operator_manifests_archive")
+            operator_archive_name = extra.get("operator_manifests_archive")
 
-        if not archive_name:
+        if not operator_archive_name:
             # No operator manifests on this build
             return
 
-        operator_archive = [a for a in archives if a["filename"] == archive_name]
+        operator_archive = [
+            a for a in archives if a["filename"] == operator_archive_name
+        ]
+        non_operator_archives = [
+            a for a in archives if a["filename"] != operator_archive_name
+        ]
         if len(operator_archive) != 1:
             message = (
                 "koji build %s metadata refers to missing operator-manifests "
                 'archive "%s"'
-            ) % (nvr, archive_name)
+            ) % (nvr, operator_archive_name)
             raise ValueError(message)
 
+        helper = ContainerArchiveHelper(meta, non_operator_archives[0])
```

Comment:
OK, so this is not actually loading any unique field for the operator manifests bundle.

What you're doing is effectively taking the 'labels' field from an arbitrarily selected one of the container images from the same build and then duplicating that on the OperatorManifestPushItem.

Why do you want to do that?  Why isn't it explained in the PR or commit message?  And why would it be correct to copy it specifically from whichever of the images appears first in this list?

All the info you're copying here is already readily available to clients of this library, e.g.

```
$ pushsource-ls brew:container_build=network-observability-operator-bundle-container-1.3.0-79
# Loaded source brew:container_build=network-observability-operator-bundle-container-1.3.0-79
ContainerImagePushItem(
    name="docker-image-sha256:b45d435ae85a20d37ad2e2ed84ccb5da81a5c7b9be68f3ce6b1823046b723167.aarch64.tar.gz",
    state="PENDING",
    src=None,
    dest=[],
    md5sum=None,
    sha256sum=None,
    origin=None,
    build="network-observability-operator-bundle-container-1.3.0-79",
    build_info=KojiBuildInfo(
        name="network-observability-operator-bundle-container",
        version="1.3.0",
        release="79",
        id=2644267,
    ),
    signing_key=None,
    dest_signing_key=None,
    source_tags=[
        "network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-aarch64"
    ],
    labels=frozendict.frozendict(
        {
            "com.redhat.component": "network-observability-operator-bundle-container",
            "com.redhat.delivery.operator.bundle": "true",
            "com.redhat.license_terms": "https://www.redhat.com/agreements",
            "com.redhat.openshift.versions": "v4.10",
        }
    ),
    arch="arm64",
    pull_info=ContainerImagePullInfo(
        tag_specs=[
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610",
                media_types=[
                    "application/vnd.docker.distribution.manifest.list.v2+json"
                ],
            ),
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-aarch64",
                media_types=[],
            ),
        ],
        digest_specs=[
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:53a98e8383cdd42588d58b64f07b73a60100d9812ce18c0357cb715eb01127cd",
                media_type="application/vnd.docker.distribution.manifest.list.v2+json",
            ),
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:ad77a544def67426fdee3d05c7b4cd389884e8f21f51ff90df76d87c2ad77748",
                media_type="application/vnd.docker.distribution.manifest.v2+json",
            ),
        ],
        media_types=[
            "application/vnd.docker.distribution.manifest.list.v2+json",
            "application/vnd.docker.distribution.manifest.v2+json",
        ],
    ),
),

ContainerImagePushItem(
    name="docker-image-sha256:0e618fb235f9ebd291a7129acc9b551f328fa40ae2ba5ae691e7227422e5dfb1.x86_64.tar.gz",
    state="PENDING",
    src=None,
    dest=[],
    md5sum=None,
    sha256sum=None,
    origin=None,
    build="network-observability-operator-bundle-container-1.3.0-79",
    build_info=KojiBuildInfo(
        name="network-observability-operator-bundle-container",
        version="1.3.0",
        release="79",
        id=2644267,
    ),
    signing_key=None,
    dest_signing_key=None,
    source_tags=[
        "network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-x86_64"
    ],
    labels=frozendict.frozendict(
        {
            "com.redhat.component": "network-observability-operator-bundle-container",
            "com.redhat.delivery.operator.bundle": "true",
            "com.redhat.license_terms": "https://www.redhat.com/agreements",
            "com.redhat.openshift.versions": "v4.10",
        }
    ),
    arch="amd64",
    pull_info=ContainerImagePullInfo(
        tag_specs=[
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610",
                media_types=[
                    "application/vnd.docker.distribution.manifest.list.v2+json"
                ],
            ),
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-x86_64",
                media_types=[],
            ),
        ],
        digest_specs=[
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:53a98e8383cdd42588d58b64f07b73a60100d9812ce18c0357cb715eb01127cd",
                media_type="application/vnd.docker.distribution.manifest.list.v2+json",
            ),
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:25b8e6c5ac870f4dce4d8bb2a478d0e0c5a232d917c96e49435585069ba31888",
                media_type="application/vnd.docker.distribution.manifest.v2+json",
            ),
        ],
        media_types=[
            "application/vnd.docker.distribution.manifest.list.v2+json",
            "application/vnd.docker.distribution.manifest.v2+json",
        ],
    ),
),

ContainerImagePushItem(
    name="docker-image-sha256:72147910fc50257218fae1d0f943bae77266117cb6024ad3f897e1b924a9b9cb.s390x.tar.gz",
    state="PENDING",
    src=None,
    dest=[],
    md5sum=None,
    sha256sum=None,
    origin=None,
    build="network-observability-operator-bundle-container-1.3.0-79",
    build_info=KojiBuildInfo(
        name="network-observability-operator-bundle-container",
        version="1.3.0",
        release="79",
        id=2644267,
    ),
    signing_key=None,
    dest_signing_key=None,
    source_tags=[
        "network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-s390x"
    ],
    labels=frozendict.frozendict(
        {
            "com.redhat.component": "network-observability-operator-bundle-container",
            "com.redhat.delivery.operator.bundle": "true",
            "com.redhat.license_terms": "https://www.redhat.com/agreements",
            "com.redhat.openshift.versions": "v4.10",
        }
    ),
    arch="s390x",
    pull_info=ContainerImagePullInfo(
        tag_specs=[
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610",
                media_types=[
                    "application/vnd.docker.distribution.manifest.list.v2+json"
                ],
            ),
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-s390x",
                media_types=[],
            ),
        ],
        digest_specs=[
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:53a98e8383cdd42588d58b64f07b73a60100d9812ce18c0357cb715eb01127cd",
                media_type="application/vnd.docker.distribution.manifest.list.v2+json",
            ),
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:85fc854d591783b909d154d562f36e37fab27dd08615260c912d9d9cd0f3223b",
                media_type="application/vnd.docker.distribution.manifest.v2+json",
            ),
        ],
        media_types=[
            "application/vnd.docker.distribution.manifest.list.v2+json",
            "application/vnd.docker.distribution.manifest.v2+json",
        ],
    ),
),

ContainerImagePushItem(
    name="docker-image-sha256:f8d1988488e036dde9cd216f38f1b393c79c10670e84c4605602b3c8f8eea7f0.ppc64le.tar.gz",
    state="PENDING",
    src=None,
    dest=[],
    md5sum=None,
    sha256sum=None,
    origin=None,
    build="network-observability-operator-bundle-container-1.3.0-79",
    build_info=KojiBuildInfo(
        name="network-observability-operator-bundle-container",
        version="1.3.0",
        release="79",
        id=2644267,
    ),
    signing_key=None,
    dest_signing_key=None,
    source_tags=[
        "network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-ppc64le"
    ],
    labels=frozendict.frozendict(
        {
            "com.redhat.component": "network-observability-operator-bundle-container",
            "com.redhat.delivery.operator.bundle": "true",
            "com.redhat.license_terms": "https://www.redhat.com/agreements",
            "com.redhat.openshift.versions": "v4.10",
        }
    ),
    arch="ppc64le",
    pull_info=ContainerImagePullInfo(
        tag_specs=[
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610",
                media_types=[
                    "application/vnd.docker.distribution.manifest.list.v2+json"
                ],
            ),
            ContainerImageTagPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                tag="network-observability-1.3.0-rhel-9-containers-candidate-79259-20230821235610-ppc64le",
                media_types=[],
            ),
        ],
        digest_specs=[
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:53a98e8383cdd42588d58b64f07b73a60100d9812ce18c0357cb715eb01127cd",
                media_type="application/vnd.docker.distribution.manifest.list.v2+json",
            ),
            ContainerImageDigestPullSpec(
                registry="registry-proxy.engineering.redhat.com",
                repository="rh-osbs/network-observability-operator-bundle",
                digest="sha256:690f4cd5489e69ca9cfde4fdf4b3a95daf6e5c4152f3d0b2b1bddee071fd2e2e",
                media_type="application/vnd.docker.distribution.manifest.v2+json",
            ),
        ],
        media_types=[
            "application/vnd.docker.distribution.manifest.list.v2+json",
            "application/vnd.docker.distribution.manifest.v2+json",
        ],
    ),
),

OperatorManifestPushItem(
    name="network-observability-operator-bundle-container-1.3.0-79/operator_manifests.zip",
    state="PENDING",
    src=None,
    dest=[],
    md5sum=None,
    sha256sum=None,
    origin=None,
    build="network-observability-operator-bundle-container-1.3.0-79",
    build_info=KojiBuildInfo(
        name="network-observability-operator-bundle-container",
        version="1.3.0",
        release="79",
        id=None,
    ),
    signing_key=None,
    related_images=[
        "registry.redhat.io/network-observability/network-observability-console-plugin-rhel9@sha256:880c0812a2cd4b7194f417e31fbdcee951ce76776fa06e33c5c9cd479a854daa",
        "registry.redhat.io/network-observability/network-observability-ebpf-agent-rhel9@sha256:e31acc12d8690c339e13b5dd22d33e707701362bdd31086c9b4c0f48cad62cc3",
        "registry.redhat.io/network-observability/network-observability-flowlogs-pipeline-rhel9@sha256:7c7c42fff63e0599bb943a9e9617b43181360d5a5254b27f368010af275cf0c3",
        "registry.redhat.io/network-observability/network-observability-rhel9-operator@sha256:fd943746840cba26996dabb47b1beb3445571fa43cb1b859fb073e4aa114b376",
        "registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:fde73ab25944b1b98f722405d5ba83af507a3ecc732157e38ea0867a541a2575",
    ],
),

# 5 item(s) found in source
```

The OperatorManifestPushItem is already unambiguously associated with ContainerImagePushItem objects from the same build which contain labels.  I don't see a reason to duplicate fields from ContainerImagePushItem onto OperatorManifestPushItem.

Reply:
Ok, I changed the MR so labels are not copied to the model, but instead there's list of related container images. I added labels in the first place because unfortunately not all labels which are relevant to the operator are present in metadata for operator_manifest.zip archive. Specifically I need it to get operator name

Reply:
Thanks, I think this approach is better.